File: Front/TS/src/infrastructure/adapters/repository/Http/HttpFriendRepository.ts
import { FriendDto } from "../../../../domain/features/RetrieveFriendsHandler";
import { IFriendRepository } from "../../../../domain/friends/IFriendRepository";
import { IHttpService } from "../../http/IHttpService";

export class HttpFriendRepository implements IFriendRepository {
    constructor(private readonly httpService: IHttpService){}
    async GetFriendsAsync(): Promise<FriendDto[]> {
        const response = await this.httpService.Get(`friendship`)
        if(response.Code == 200) {
            return Promise.resolve(response.Data as FriendDto[]);
        }
        return Promise.reject();
    }
    async SaveAsync(memberId: string): Promise<void> {
        const response = await this.httpService.Post(`friendship/${memberId}`)
        if(response.Code == 200) {
            return Promise.resolve(undefined!);
        }
        return Promise.reject(response.Error);
    }

}

File: Front/.Net Maui/BetFriends.Infrastructure/Repository/Http/HttpFriendRepository.cs
using BetFriend.Domain.Friends;
using BetFriends.Domain.Features.RetrieveFriends;

namespace BetFriends.Infrastructure.Repository.Http;

internal class HttpFriendRepository(IHttpClientFactory httpClientFactory) : IFriendRepository
{
    private readonly HttpClient httpClient = httpClientFactory.CreateClient("bets");
    public async Task AddAsync(string id, CancellationToken cancellationToken)
    {
        var response = await httpClient.PostAsync($"/friendship/{id}", null);
        response.EnsureSuccessStatusCode();
    }

    public Task<IReadOnlyCollection<FriendDto>> GetFriendsAsync()
    {
        throw new NotImplementedException();
    }
}
